src/deployer/solana.ts

```diff
@@ -0,0 +1,181 @@
+import { Program, type Provider } from "@coral-xyz/anchor"
+import { TOKEN_PROGRAM_ID } from "@solana/spl-token"
+import {
+  Keypair,
+  PublicKey,
+  SYSVAR_CLOCK_PUBKEY,
+  SYSVAR_RENT_PUBKEY,
+  SystemProgram,
+} from "@solana/web3.js"
+import type { MPCSignature, TokenMetadata } from "../types"
+import type { BridgeTokenFactory } from "../types/solana/bridge_token_factory"
+import * as BridgeTokenFactoryIdl from "../types/solana/bridge_token_factory.json"
+
+const MPL_PROGRAM_ID = new PublicKey("metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s")
+
+export class SolanaDeployer {
+  private readonly wormholeProgramId: PublicKey
+  private readonly program: Program<BridgeTokenFactory>
+
+  private static readonly SEEDS = {
+    CONFIG: new Uint8Array([99, 111, 110, 102, 105, 103]),
+    AUTHORITY: new Uint8Array([97, 117, 116, 104, 111, 114, 105, 116, 121]),
+    WRAPPED_MINT: new Uint8Array([119, 114, 97, 112, 112, 101, 100, 95, 109, 105, 110, 116]),
+    VAULT: new Uint8Array([118, 97, 117, 108, 116]),
+  }
+
+  constructor(provider: Provider, wormholeProgramId: PublicKey) {
+    this.wormholeProgramId = wormholeProgramId
+    this.program = new Program(BridgeTokenFactoryIdl as BridgeTokenFactory, provider)
+  }
+
+  private configId(): [PublicKey, number] {
```

Comment:
I'm not sure if `Id` is the term used in solana for account identifiers, but I could be wrong. I would probably go with `configAddress` or `config`